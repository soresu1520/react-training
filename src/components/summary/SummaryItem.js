import styled from "styled-components";

const SummaryItem = ({ cartItem }) => {
  return (
    <SummaryItemDiv data-testid="summary-item">
      <Image src={`/assets/food/${cartItem.image}`} alt={cartItem.name} />
      <ItemText>{cartItem.name}</ItemText>
      <ItemTextSmallBasis>x{cartItem.quantity}</ItemTextSmallBasis>
      <ItemTextRight>{(cartItem.quantity * cartItem.price).toFixed(2)} $</ItemTextRight>
    </SummaryItemDiv>
  );
};

export default SummaryItem;

const SummaryItemDiv = styled.div`
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  height: 4em;
  width: 100%;
  padding: 0.5em 0.7em;

  @media (max-width: 550px) {
    flex-flow: column nowrap;
    height: 7em;
    flex-basis: 40%;
  }
`;

const ItemText = styled.h5`
  font-size: var(--font-size-h5);
  font-weight: var(--font-weight-heading);
  color: var(--dark-icon);
  margin-top: 0;
  margin-bottom: 0;
  flex-basis: 30%;
`;

const ItemTextSmallBasis = styled(ItemText)`
  flex-basis: 20%;
  text-align: center;
`;

const ItemTextRight = styled(ItemTextSmallBasis)`
  flex-basis: 20%;
  text-align: right;
`;

const Image = styled.img`
  height: 100%;
  width: auto;
  margin: 0;
  padding-right: 0.7em;

  @media (max-width: 550px) {
    height: 50%;
  }
`;
