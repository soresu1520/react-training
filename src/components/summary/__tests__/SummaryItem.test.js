import renderer from "react-test-renderer";
import SummaryItem from "../SummaryItem";

const cartItem = { productId: 2, name: "Pancakes", price: 5, image: "pancakes.png", quantity: 1 };

it("summary item renders correctly", () => {
  const tree = renderer.create(<SummaryItem cartItem={{ cartItem }} />).toJSON();
  expect(tree).toMatchSnapshot();
});
