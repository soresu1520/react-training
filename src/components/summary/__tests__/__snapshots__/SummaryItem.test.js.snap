// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summary item renders correctly 1`] = `
<div
  className="sc-bczRLJ jaAqGV"
  data-testid="summary-item"
>
  <img
    className="sc-eCYdqJ eYzrYi"
    src="/assets/food/undefined"
  />
  <h5
    className="sc-gsnTZi bLgvay"
  />
  <h5
    className="sc-gsnTZi sc-dkzDqf bLgvay isIQSg"
  >
    x
  </h5>
  <h5
    className="sc-gsnTZi sc-dkzDqf sc-hKMtZM bLgvay isIQSg dIhHBk"
  >
    NaN
     $
  </h5>
</div>
`;
